url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/TestForHttpRequest.java

```diff
@@ -0,0 +1,114 @@
+package net.hetimatan.util.http;
+
+import net.hetimatan.io.file.MarkableFileReader;
+import net.hetimatan.io.filen.CashKyoroFile;
+import net.hetimatan.util.http.HttpRequestHeader;
+import net.hetimatan.util.http.HttpObject;
+import net.hetimatan.util.http.HttpRequestLine;
+import net.hetimatan.util.http.HttpRequest;
+import java.io.IOException;
+
+import junit.framework.TestCase;
+
+public class TestForHttpRequest extends TestCase {
+
+	public void testEncode001() throws IOException {
+		HttpRequest uri = new HttpRequest(
+				HttpRequestLine.decode("GET /announce HTTP/1.1\r\n"));
+		CashKyoroFile output = new CashKyoroFile(512);
+		uri.encode(output.getLastOutput());
+
+		output.seek(0);
+		byte[] buffer = new byte[(int)output.length()];
+		int len = output.read(buffer);
+		String tag = new String(buffer, 0, len);
+		assertEquals("GET /announce HTTP/1.1\r\n\r\n", tag);
+	}
+
+	public void testEncode002() throws IOException {
+		HttpRequest uri = new HttpRequest(HttpRequestLine.decode("GET /announce HTTP/1.1\r\n"));
+		uri.addHeader(HttpRequestHeader.decode("User-agent: Mozilla/5.0 (Linux; U; Android 2.2.1; ja-jp; Full Android Build/MASTER) AppleWebKit/533.1 (KHTML, like Gecko) Version/4.0 Mobile Safari/533.1\r\n"));
+		CashKyoroFile output = new CashKyoroFile(512);
+		uri.encode(output.getLastOutput());
+
+		output.seek(0);
+		byte[] buffer = new byte[(int)output.length()];
+		int len = output.read(buffer);
+		String tag = new String(buffer, 0, len);
+		assertEquals("GET /announce HTTP/1.1\r\n"
+		+"User-agent: Mozilla/5.0 (Linux; U; Android 2.2.1; ja-jp; Full Android Build/MASTER) AppleWebKit/533.1 (KHTML, like Gecko) Version/4.0 Mobile Safari/533.1\r\n"
+		+"\r\n", tag);
+	}
+
+	public void testEncode003() throws IOException {
+		HttpRequest uri = new HttpRequest(HttpRequestLine.decode("GET /announce HTTP/1.1\r\n"));
+		uri.addHeader(HttpRequestHeader.decode("Accept-encoding: gzip\r\n"));
+		uri.addHeader(HttpRequestHeader.decode("Host: 127.0.0.1:6969\r\n"));
+		uri.addHeader(HttpRequestHeader.decode("User-agent: BitTorrent/4.0.2\r\n"));
+		CashKyoroFile output = new CashKyoroFile(512);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *output*. Currently, there are execution paths that do not contain closure statements, for example, when HttpRequest.encode throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).